yt_dlp/downloader/fragment.py

```diff
@@ -21,6 +23,77 @@
 )
 
 
+class Timer:
+    TIMER_FUNC = time.monotonic
+
+    def __init__(self):
+        self._last_ts = self.TIMER_FUNC()
+
+    def elapsed_seconds(self) -> float:
+        return self.TIMER_FUNC() - self._last_ts
+
+    def has_elapsed(self, seconds: float) -> bool:
```

Comment:
I'm not sure I like the side effect of this function moving along the timer if it would return true. If you just want to check if it has elapsed, but act apon it later this would prevent that. I'd rather have it be explicit.

For example, `if self.timer.has_elapsed(seconds=self.UPDATE_TIMESPAN_S):` from `Speedometer` calls it, moving the timer forward, but that shouldn't be happening right?